CRM/Admin/Form/Options.php

```diff
@@ -174,11 +178,12 @@ public function buildQuickForm() {
         ['CRM_Core_DAO_OptionValue', $this->_id, $this->_gid, 'value', $this->_domainSpecific]
       );
     }
-    else {
+
+    // Add icon & color if this option group supports it.
+    if ($optionGroup['shape'] && in_array('icon', $optionGroup['shape'])) {
       $this->add('text', 'icon', ts('Icon'), ['class' => 'crm-icon-picker', 'title' => ts('Choose Icon'), 'allowClear' => TRUE]);
     }
-
-    if (in_array($this->_gName, ['activity_status', 'case_status'])) {
+    if ($optionGroup['shape'] && in_array('color', $optionGroup['shape'])) {
```

Comment:
This is the upshot of the change for our forms, we no longer need to hard-code a list of `OptionGroup`s that support "color" or "icon" we can rely on the metadata.